consensus/parlia/parlia.go

```diff
@@ -751,6 +764,36 @@ func (p *Parlia) snapshot(chain consensus.ChainHeaderReader, number uint64, hash
 	return snap, err
 }
 
+func (p *Parlia) findSnapFromHistorySegment(hash common.Hash, number uint64) (*Snapshot, bool) {
+	if p.hsm == nil {
+		return nil, false
+	}
+	segment := p.hsm.CurSegment(number + 1)
```

Comment:
number+1? related GetConsensusData

Reply:
Yes, the segment's consensus data is the previous block snapshot, so it can provide segment start block validation in parlia.